Differentiate between lib and lib64 in modules

kgerheiser opened the pull request:
Instead of hardcoding `lib`, make it a templated value and at module installation detect `CMAKE_INSTALL_LIBDIR` and fill it into the module using `configure_file`.

kgerheiser committed:
Differentiate between lib and lib64 in modules by using CMAKE_INSTALL_LIBDIR

aerorahul commented on stack_helpers.sh:
```diff
@@ -29,8 +29,21 @@ function update_modules {
-  $SUDO cp $tmpl_file $version.lua
 
+  # CMAKE_INSTALL_LIBDIR is used by some projects (i.e. lib, lib64)
+  # Detect this when installing the module so the module variables point to the correct path
```
can we create that `CMakeLists.txt` file here instead of adding a new file to the repo.  Its purpose is really internal here.

aerorahul commented:
I like the changes. 
But, I have a question.
I see you are relying on `cmake` to obtain `CMAKE_INSTALL_LIBDIR` and install the modulefile.
What do you do when cmake itself is being built and deployed as a module?

kgerheiser committed:
Make sure CMake is available before using it

Check if building MPI, compiler, or CMake and if so don't run configure_file on module

Move CMake script into generate file

kgerheiser commented:
@aerorahul I added a check if building `cmake`, `gnu`, or `mpi` (the first three things built) and for those just copy the module over, otherwise CMake is available.

aerorahul commented on stack_helpers.sh:
```diff
@@ -29,8 +29,34 @@ function update_modules {
+
+    # Install the module with configure_file, replacing ${CMAKE_INSTALL_LIBDIR} (and potentially other variables)
+    # with the actual value for that system
+    $SUDO cmake -DCMAKE_INSTALL_LIBDIR=${CMAKE_INSTALL_LIBDIR} \
```
This is good.
I would just note that we will need more template variables e.g. `PYTHON_VERSION` in `bufr.lua`.
We can address that in a later PR.

aerorahul reviewed (approved):
looks good.
You will need to bring you branch up to date before CI and merge.

kgerheiser committed:
Merge branch 'develop' into bugfix/module-cmake-install-dir

kgerheiser committed:
Merge branch 'develop' into bugfix/module-cmake-install-dir

kgerheiser committed:
Create module for NCIO

kgerheiser committed:
Merge branch 'bugfix/module-cmake-install-dir' of github.com:kgerheiser/hpc-stack into bugfix/module-cmake-install-dir
